compiler/packages/babel-plugin-react-compiler/src/CompilerError.ts

```diff
@@ -454,31 +421,58 @@ export class CompilerError extends Error {
     return this.hasErrors() ? Err(this) : Ok(undefined);
   }
 
-  /*
-   * An error is critical if it means the compiler has entered into a broken state and cannot
-   * continue safely. Other expected errors such as Todos mean that we can skip over that component
-   * but otherwise continue compiling the rest of the app.
-   */
-  isCritical(): boolean {
-    return this.details.some(detail => {
-      switch (detail.severity) {
-        case ErrorSeverity.Invariant:
-        case ErrorSeverity.InvalidJS:
-        case ErrorSeverity.InvalidReact:
-        case ErrorSeverity.InvalidConfig:
-        case ErrorSeverity.UnsupportedJS:
-        case ErrorSeverity.IncompatibleLibrary: {
-          return true;
-        }
-        case ErrorSeverity.CannotPreserveMemoization:
-        case ErrorSeverity.Todo: {
-          return false;
-        }
-        default: {
-          assertExhaustive(detail.severity, 'Unhandled error severity');
-        }
+  /**
+   * Returns true if any of the error details are of severity Error.
+   */
+  isError(): boolean {
+    let res = false;
+    for (const detail of this.details) {
+      if (detail.severity === ErrorSeverity.Off) {
+        return false;
```

Comment:
why not just ignore details that are off? maybe instead of returning a boolean, we could have `CompilerError.proto.errors(): Array<CompilerDetail> | null` and that filters out any details that are off, and returns null if empty?

Reply:
Addressed in #34409